.github/workflows/build.yml

```diff
@@ -7,44 +7,175 @@ on:
   push:
     branches:
       - main
+    tags:
+      - '*'
+  workflow_dispatch:
 
 jobs:
-  release:
-    runs-on: ${{ matrix.os }}
-
+  build:
     strategy:
       matrix:
-        os: [macos-10.15, windows-latest, ubuntu-latest]
+        include:
+          - runner: macos-11
+            os: darwin
+            arch: amd64
+          - runner: windows-2019
+            os: windows
+            arch: amd64
+          - runner: ubuntu-20.04
+            os: linux
+            arch: amd64
```

Comment:
do you think it makes sense to add older/newer versions of linux distros to make sure we have a range of glibc versions covered?

Reply:
Unfortunately, there's no older version than Ubuntu 20.04 and MacOS 11 for GitHub-hosted runners and just ubuntu-latest for buildjet runners. All these runners currently use GCC 10. Binaries compiled with an older GCC version are compatible to newer ones (they are forward-compatible), so choosing the "oldest" available seemed to be the best approach.
Here's the support policy for GitHub Runners: https://github.com/actions/runner-images#software-and-image-support
They basically always support the last three major versions of the GCC toolchain.